dask_ml/model_selection/_successive_halving.py

```diff
@@ -0,0 +1,239 @@
+
+    model_history_ : dict of lists of dict
+        A dictionary of each models history. This is a reorganization of
+        ``history_``: the same information is present but organized per model.
+
+        This data has the structure  ``{model_id: hist}`` where ``hist`` is a
+        subset of ``history_`` and ``model_id`` are model identifiers.
+
+    history_ : list of dicts
+        Information about each model after each ``partial_fit`` call. Each dict
+        the keys
+
+        * ``partial_fit_time``
+        * ``score_time``
+        * ``score``
+        * ``model_id``
+        * ``params``
+        * ``partial_fit_calls``
+
+        The key ``model_id`` corresponds to the ``model_id`` in ``cv_results_``.
+        This list of dicts can be imported into Pandas.
+
+    best_estimator_ : BaseEstimator
+        The model with the highest validation score among all the models
+        retained by the "inverse decay" algorithm.
+
+    best_score_ : float
+        Score achieved by ``best_estimator_`` on the vaidation set after the
+        final call to ``partial_fit``.
+
+    best_index_ : int
+        Index indicating which estimator in ``cv_results_`` corresponds to
+        the highest score.
+
+    best_params_ : dict
+        Dictionary of best parameters found on the hold-out data.
+
+    scorer_ :
+        The function used to score models, which has a call signature of
+        ``scorer_(estimator, X, y)``.
+
+    n_splits_ : int
+        Number of cross validation splits.
+
+    multimetric_ : bool
+        Whether this cross validation search uses multiple metrics.
+
+
+    """
+
+    def __init__(
+        self,
+        estimator,
+        param_distribution,
+        n_initial_parameters=15,
+        start_iter=9,
+        adaptive_max_iter=None,
+        aggressiveness=3,
+        max_iter=100,
+        test_size=None,
+        patience=False,
+        tol=1e-3,
+        random_state=None,
+        scoring=None,
+    ):
+        self.estimator = estimator
+        self.param_distribution = param_distribution
+
+        self.n_initial_parameters = n_initial_parameters
+        self.start_iter = start_iter
+        self.aggressiveness = aggressiveness
+        self.adaptive_max_iter = adaptive_max_iter
+
+        super(SuccessiveHalvingSearchCV, self).__init__(
+            estimator,
+            param_distribution,
+            n_initial_parameters=n_initial_parameters,
+            max_iter=max_iter,
+            test_size=test_size,
+            patience=patience,
+            tol=tol,
+            random_state=random_state,
+            scoring=scoring,
+        )
+
+    def fit(self, X, y, **fit_params):
+        # Used to track the number of times `_adapt` has been called.
+        # Probably not required but simple
+        self._steps = 0
+
+        # Used to record the number of partial_fit calls per model.
+        # Required to ensure Hyperband records/passes parameters accurately
+        self._pf_calls = {}
```

Comment:
`_pf_calls` -> `_partial_fit_calls` or

Reply:
It seems like we're overriding `_fit` here just to set a bit of mutable state, which concerns me. IIUC the two pieces of state are

1. The number of calls to partial fit (per model) as of this call to `additional_calls`
2. The number of times adapt has been called.

Keeping this as state on the object concerns me a bit as it's now spread among two places: `self` and the `info` dict we pass around. Are either of these redundant with any information in `info`?

Reply:
I've removed `_pf_calls`, and moved `self._steps` inside `_adapt`. Is that sufficient? `_steps` probably can be calculated from `info` (and all the other parameters of `SuccessiveHalvingSearchCV`), but it seems complicated.